cylc/flow/task_pool.py

```diff
@@ -312,12 +312,17 @@ def compute_runahead(self, force=False) -> bool:
         changed (needed if max_future_offset changed, or on reload).
         """
         points: List['PointBase'] = []
+        sequence_points: Set['PointBase']
         if not self.main_pool:
             # Start at first point in each sequence, after the initial point.
-            points = list({
-                seq.get_first_point(self.config.start_point)
-                for seq in self.config.sequences
-            })
+            points = [
+                point
+                for point in {
+                    seq.get_first_point(self.config.start_point)
+                    for seq in self.config.sequences
+                }
+                if point is not None
```

Comment:
The problem was caused by a `None` value in this list. Due to the logic lower down, this caused the runahead limit point to be set to `None`.